src/NzbDrone.Core/ImportLists/Simkl/SimklImportBase.cs

```diff
@@ -0,0 +1,183 @@
+            : base(httpClient, importListStatusService, configService, parsingService, logger)
+        {
+            _importListRepository = netImportRepository;
+        }
+
+        public override IList<ImportListItemInfo> Fetch()
+        {
+            Settings.Validate().Filter("AccessToken", "RefreshToken").ThrowOnError();
+            _logger.Trace($"Access token expires at {Settings.Expires}");
+
+            // Simkl doesn't currently expire access tokens, but if they start lets be prepared
+            if (Settings.RefreshToken.IsNotNullOrWhiteSpace() && Settings.Expires < DateTime.UtcNow.AddMinutes(5))
+            {
+                RefreshToken();
+            }
+
+            var lastFetch = _importListStatusService.GetLastSyncListInfo(Definition.Id);
+            var lastActivity = GetLastActivity();
+
+            // Check to see if user has any activity since last sync, if not return empty to avoid work
+            if (lastFetch.HasValue && lastActivity < lastFetch.Value.AddHours(-2))
+            {
+                return new List<ImportListItemInfo>();
+            }
+
+            var generator = GetRequestGenerator();
+
+            return FetchItems(g => g.GetListItems(), true);
+        }
+
+        public override IParseImportListResponse GetParser()
+        {
+            return new SimklParser();
+        }
+
+        public override object RequestAction(string action, IDictionary<string, string> query)
+        {
+            if (action == "startOAuth")
+            {
+                var request = new HttpRequestBuilder(OAuthUrl)
+                    .AddQueryParam("client_id", ClientId)
+                    .AddQueryParam("response_type", "code")
+                    .AddQueryParam("redirect_uri", RedirectUri)
+                    .AddQueryParam("state", query["callbackUrl"])
+                    .Build();
+
+                return new
+                {
+                    OauthUrl = request.Url.ToString()
+                };
+            }
+            else if (action == "getOAuthToken")
+            {
+                return new
+                {
+                    accessToken = query["access_token"],
+                    expires = DateTime.UtcNow.AddSeconds(int.Parse(query["expires_in"])),
+                    refreshToken = query["refresh_token"],
+                    authUser = GetUserId(query["access_token"])
+                };
+            }
+
+            return new { };
+        }
+
+        private DateTime GetLastActivity()
+        {
+            var request = new HttpRequestBuilder(string.Format("{0}/sync/activities", Settings.BaseUrl)).Build();
+
+            request.Headers.Add("simkl-api-key", ClientId);
+            request.Headers.Add("Authorization", "Bearer " + Settings.AccessToken);
+
+            try
+            {
+                var response = _httpClient.Get<SimklSyncActivityResource>(request);
+
+                if (response != null && response.Resource != null)
```

Comment:
Could shorten this to `response?.Resource != null`, probably worth getting into that habbit.